syntax = "proto3";

package notification_service;

service NotificationSender {
  rpc send_notification(Notification) returns (NotificationResponse) {}

}

message Notification {
  int32 id = 1;
  int32 user_id = 2;
  NotificationType type = 3;
  string title = 4;
  string message = 5;
}

enum NotificationType {
  EVENT_REMINDER = 0;
  BOOKING_CONFIRMATION = 1;
  EVENT_CANCELLED = 2;
  EVENT_UPDATED = 3;
}

message NotificationResponse {
  bool success = 1;
  string message = 2;
  Notification notification = 3;
}
